docs/transaction-verification.md

```diff
@@ -0,0 +1,169 @@
+that the node does not know about the block of which the identity provider was added to the chain. 
+The transaction will be added to the transaction table and re-verified in `pre execution` verification.
+4. `existence of the anonymity revokers` As with the existence of valid identity providers there must also be valid anonymity revokers present in 
+the credential deployment when the transaction was received individually, otherwise it will be rejected. But if the transaction was received as part of a 
+block then it might be valid as the anonymity revoker(s) could've been added in a future block.
+The transaction will be added to the transaction table and re-verified in `pre execution` verification.
+5. `cryptographic verification` Finally the signatures along with the `credential proof` on the credential deployment will be verified and if they are invalid then 
+it will be rejected and not relayed to `consensus`. This is always the case as identity providers and anonymity revokers have no means 
+to update or modify the on-chain key pair.
+ 
+Successfully verified credential deployments will be put into the `Transaction Table` and later be executed by the `Scheduler`.
+The `Scheduler` asserts that the credential deployment transaction is valid - by asserting the above requirements - before executing it.
+ 
+#### Chain Update
+Chain updates are checked for: 
+ 
+1. `valid effective time` A chain update has an `effective time` and a `timeout` associated. While the latter indicates when the current 
+chain update should no longer be taken into consideration and the former is a timestamp of when the chain update should be 'active' from.
+It is verified that the timeout is not later than the effective time or is immediate. If this is the case then the transaction will always be rejected. 
+2. `disallowing old sequence numbers` Chain updates each have an associated `sequence number`. This number serves as a `nonce`, hence each chain update must have a unique 
+`sequence number`. Thus Chain updates with an associated `sequence number` that has already been used are always rejected and not forwarded to `consensus`.
+3. `signature verification` A chain update must be correctly signed by the authorized signer(s). If this is not the case i.e., the signature is invalid 
+then the transaction is rejected and never passed to `consensus` if the transaction was received individually. 
+However if the transaction was part of a block the transaction will not immediately be rejected as the set of authorized keys can be changed over time. 
+ 
+Successfully verified chain updates are put in the transaction table and later executed by the `Scheduler`.
+The `Scheduler` is making sure that the above requirements are satisfied.
+ 
+#### Transaction
+Transactions are checked for:
+ 
+1. `energy supply` If the sender did not supply enough energy for the cost of the transaction then it will always be rejected.
```

Comment:
Note that we only check the "check header cost" in this pre-verification. Transaction can still be included with "out of energy" if it passes this check.